packages/api/src/hooks/useCreateActivityStatusRenderer.tsx

```diff
@@ -13,32 +14,48 @@ const {
   ActivityClientState: { SEND_FAILED, SENDING, SENT }
 } = Constants;
 
-const ActivityStatusContainer = ({ activity, hideTimestamp, nextVisibleActivity }) => {
+type ActivityStatusContainerProps = {
+  activity: DirectLineActivity;
+  hideTimestamp: boolean;
+  nextVisibleActivity: DirectLineActivity;
+};
+
+const ActivityStatusContainer: VFC<ActivityStatusContainerProps> = ({
+  activity,
+  hideTimestamp,
+  nextVisibleActivity
+}) => {
   const { activityStatusRenderer: createActivityStatusRenderer } = useWebChatAPIContext();
   const getSendTimeoutForActivity = useGetSendTimeoutForActivity();
 
   // SEND_FAILED from the activity is ignored, and is instead based on styleOptions.sendTimeout.
   // Note that the derived state is time-sensitive. The useTimePassed() hook is used to make sure it changes over time.
   const {
-    channelData: { clientTimestamp = 0, state } = {},
     from: { role }
-  }: {
-    channelData: {
-      clientTimestamp?: number;
-      state?: SendState;
-    };
-    from: {
-      role: string;
-    };
-  } = activity;
-
-  const activitySent = state !== SENDING && state !== SEND_FAILED;
+  }: DirectLineActivity = activity;
+
   const fromUser = role === 'user';
-  const sendTimeout = getSendTimeoutForActivity({ activity });
+  let activitySent: boolean;
+  let sendTimeoutAt: number;
+
+  if (fromUser) {
```

Comment:
This is pure logic expansion, due to limitations in TypeScript narrowing capability.